iothub/device/src/InternalClient.cs

```diff
@@ -1887,9 +1890,22 @@ internal void ValidateModuleTransportHandler(string apiName)
 
         #region Convention driven operations
 
-        internal Task<ClientProperties> GetPropertiesAsync(PayloadConvention payloadConvention, CancellationToken cancellationToken)
+        /// <summary>
+        /// Creates the correct <see cref="IWritablePropertyResponse"/> to be used with this serializer
+        /// </summary>
+        /// <param name="value">The value of the property.</param>
+        /// <param name="statusCode">The status code of the write operation.</param>
+        /// <param name="version">The version the property is responding to.</param>
+        /// <param name="description">An optional description of the writable property response.</param>
+        /// <returns></returns>
+        internal IWritablePropertyResponse CreateWritablePropertyResponse(object value, int statusCode, long version, string description = default)
         {
-            payloadConvention ??= DefaultPayloadConvention.Instance;
+            return _clientOptions.PayloadConvention.CreateWritablePropertyResponse(value, statusCode, version, description);
```

Comment:
Is there an advantage of providing this method on the device client/ module client? One that I can think of is for scenarios where a user doesn't set any convention, then they don't even need to know about the existence of a convention but can retrieve the individual properties (serializer, encoder) or call this method from the device client. I like that, but one potential drawback I see is that we'll need to keep adding any new methods/ properties we add to the convention to the clients as well.

What do you think about highlighting the existence of a convention (and the default), so that a user can call these methods/ fetch the properties from the convention object itself; without us having to duplicate these getters?